### DISCUSSION TITLE
getShopReceipt in API v3, how to pass a list of order ids.

### DISCUSSION THREAD

**Discussion by Trakis** [etsy/open-api#666] [Q&A] [answered]
getShop_Receipt2 <API v2> replaced by getShopReceipt <API v3>,   method  "getShop_Receipt2"<API v2>  was accepting parameter "receipt_id" as type array([int]), but getShopReceipt <API v3> only takes one receipt_id, not a list. Does there is an option or it will become available where I could extract receipts by providing a list of receipt ids?

**Comment by etsyachristensen** [comment, marked as answer]
At this time, no there is no option for that, nor any plan to add that. There is the (getShopReceipts)[https://developers.etsy.com/documentation/reference#operation/getShopReceipts] endpoint that will retrieve a list of receipts based on several filters and sort options.

**Comment by ablagoev** [comment]
Hey,

This missing feature is really disappointing. Given that you do not return cancelled orders with the min_modified filters, the only way to check for cancelled orders is checking them by id.

So you have two endpoints, that are inadequate for a very common use case and people will needlessly spam your endpoints receipt by receipt. There are a lot of other additional use cases where getting multiple receipts by id is very useful. I can't imagine what your reasoning was for not implementing this in the huge timeframe for v3.

Not to mention your current documentation for the getShopReceipts says:

```
Requests the Shop Receipts from a specific Shop, unfiltered or filtered by receipt id range or offset, date, paid, and/or shipped purchases.
```

and yet there is no mention of the actual parameter further below and it contradicts what you have been saying in this issue.

Finally the getShopReceipts actually does not respect the limit and offset parameters and it also does not apply the default limit, so at the moment if you request the getShopReceipts endpoint all historical orders are returned for the shop in a single request/response. So in addition to the lacking functionality, it is also buggy.

The developer experience was not good with v2, and with v3 it has gotten even worse. The new API offers no benefits at all to your sellers, it only creates further costs for them and will be a lot slower (your are forcing devs to make a lot more requests than what previously was necessary).
